src/main/java/ch/uzh/ifi/hase/soprafs22/game/GameDelta.java

```diff
@@ -9,10 +9,16 @@ public class GameDelta {
     private final Map<Position, Integer> unitHealths;
     private final TurnInfo turnInfo;
 
-    public GameDelta(MoveCommand moveCommand, TurnInfo turnInfo, Map<Position, Integer> unitHealths) {
+    private final GameOverInfo gameOverInfo;
+
+    private final SurrenderInfo surrenderInfo;
+
+    public GameDelta(MoveCommand moveCommand, TurnInfo turnInfo, Map<Position, Integer> unitHealths, GameOverInfo gameOverInfo, SurrenderInfo surrenderInfo) {
```

Comment:
Sorry @hilalcomak ,
I'm nitpicking again, but since our constructor starts to bloat:
The only two elements our GameDelta object needs to be well defined are MoveCommand and TurnInfo, right?
UnitHealths is sometimes null (when we don't make an attack, just a move), GameOverInfo is null when the game isn't over, analogous for the SurrenderInfo or am I mistaking?

Therefore maybe it's a good idea to split it up into a constructor:
public GameDelta(MoveCommand moveCommand, TurnInfo turnInfo, Map<Position, Integer> unitHealths){
this.turnInfo = turnInfo;
        this.moveCommand =  moveCommand;
        this.unitHealths = null;
        this.gameOverInfo = null;
        this.surrenderInfo = null;
}

and 3 setters for unitHealths, gameOverInfo and surrenderInfo that will be used on demand.

It's just style and my idea but of course I'm open to yours.

Reply:
In the client we check if things are not null and then just update what we get.
So you can also set everything to null by default...

TurnInfo should also be able to be null, I think we just set it when we change the turn, right?

Reply:
Ah yes, you‘re right.
But moveCommand should never be null at least.

in that case, the constructor with 1 parameter and 4 setters would be my proposal.

@hilalcomak knows better than me what can be null and what not anyway ;)

Reply:
MoveCommand can be null when users surrender. I will have all default to null then.

Reply:
Addressed the comments, does it look fine now?

Reply:
As mentioned in  #324, I went back to no setters and use multiple constructors.  I've already changed 3 times back and forth so I'm merging it: @paolovic

Reply:
> As mentioned in #324, I went back to no setters and use multiple constructors. I've already changed 3 times back and forth so I'm merging it. @paolovic

That's why I had a guilty conscience and opened an extra branch, removed the setters for you, made a pull request to your branch and asked you politely for your review (twice). And made by the way also other changes besides removing your setters (which returned "this" (???))
Now you couldn't wait for 15 Minutes. 
Cool way of cooperation.
Thank you very much 👍🏻